### PR TITLE
Added note to recipient for desktop client.

### PR DESCRIPTION
**Pull request by IzabelaBakollari**
Added a textbox similar to server version to send a note on share.

Refered to the issue https://github.com/nextcloud/desktop/issues/1139

### CODE REVIEW

**Comment by camilasan** [comment]
Please, add screenshots when there are UI changes.

**Comment by jancborchardt** [comment]
Yep, as soon as you add screenshots I can check it out. :)

**Comment by IzabelaBakollari** [comment]
These are the UI changes :)

Screenshot_20190325_174728

File: src/libsync/capabilities.cpp [hunk 1]
```diff
@@ -74,6 +74,16 @@ int Capabilities::sharePublicLinkExpireDateDays() const
...
 }
 
+QString Capabilities::sharePublicLinkNote() const
+{
+    return _capabilities["files_sharing"].toMap()["public"].toMap()["note"].toString();
+}
+
+bool Capabilities::sharePublicLinkEnforceNote() const
+{
+    return _capabilities["files_sharing"].toMap()["public"].toMap()["note"].toMap()["enforced"].toBool();
```
Comment (rullzer) [inline src/libsync/capabilities.cpp]: I do not think we have this enforcement on the server.

File: src/libsync/capabilities.cpp [hunk 2]
```diff
@@ -74,6 +74,16 @@ int Capabilities::sharePublicLinkExpireDateDays() const
     return _capabilities["files_sharing"].toMap()["public"].toMap()["expire_date"].toMap()["days"].toInt();
 }
 
+QString Capabilities::sharePublicLinkNote() const
+{
+    return _capabilities["files_sharing"].toMap()["public"].toMap()["note"].toString();
```
Comment (rullzer) [inline src/libsync/capabilities.cpp]: I Don't think this exists either. I guess it is a version check kind of thing.

File: src/libsync/capabilities.h [hunk 3]
```diff
@@ -125,6 +126,8 @@ class OWNCLOUDSYNC_EXPORT Capabilities
     /**
      * Whether conflict files should remain local (default) or should be uploaded.
      */
+      QString sharePublicLinkNote() const;
```
Comment (rullzer) [inline src/libsync/capabilities.h]: Now the comment is no longer valid ;)

File: src/gui/sharemanager.h [hunk 4]
```diff
@@ -140,8 +140,7 @@ class LinkShare : public Share
         const Permissions permissions,
         bool passwordSet,
         const QUrl &url,
-        const QDate &expireDate);
-
+        const QDate &xpireDate);
```
Comment (rullzer) [inline src/gui/sharemanager.h]: Typo I guess

File: src/gui/sharemanager.cpp [hunk 5]
```diff
@@ -192,6 +197,25 @@ void LinkShare::setName(const QString &name)
...
+{
+    OcsShareJob *job = new OcsShareJob(_account);
+    connect(job, &OcsShareJob::shareJobFinished, this, &LinkShare::slotNoteSet);
+    connect(job, &OcsJob::ocsError, this, &LinkShare::slotOcsError);
+    job->setNote(getId(), note);
+}
+
+void LinkShare::slotNoteSet(const QJsonDocument &, const QVariant &note)
+{
+    _note = note.toString() != "";
```
Comment (rullzer) [inline src/gui/sharemanager.cpp]: This tries to set _note to a boolean right?

**Comment by jancborchardt** [comment]
@IzabelaBakollari could you reduce the note field height to 3 lines by default – if possible that it automatically expands based on content. Otherwise looks good!

**Comment by DominiqueFuchs** [comment]
Note: Continued in #1429 as this branch is forked and @IzabelaBakollari is not around anymore based on her other PR (which is totally understandable, no offense - quite some time since initial PR)

**Pull request closed without merging**